sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -3595,4 +3647,61 @@ function argumentsToString<A extends Args[] | { [argName: string]: any } = [] |
   return output;
 }
 
+/**
+ * Converts an object of arguments to an array string representation,
+ * preserving the position-based structure of the arguments.
+ *
+ * @template A - Type parameter representing the type of the arguments.
+ * @param {A} args - The arguments to convert.
+ * @returns {string} - The string representation of the arguments.
+ */
+function argumentsToPositionString<A extends any[] | { [argName: string]: any } = [] | {}>(args: A): string {
+  let output = '';
+  function printObject(obj: any) {
+    Object.keys(obj).forEach(key => {
+      const value = obj[key];
+
+      if (Array.isArray(value)) {
+        output += `[`;
+        printArray(value);
+        output += `],`;
+      } else if (typeof value === 'object') {
+        //value is a Local or Parameter
+        if (value instanceof Variable) {
+          output += `${value.toReferenceString()},`;
+        } else {
+          output += `[`;
+          printValue(value);
+          output += `],`
+        }
+      } else {
+        output += `${typeof value === 'string' ? `'${value}'` : value},`;
+      }
+    });
+  }
+
+  function printArray(array: any[]) {
+    array.forEach((item: any) => {
+      output += `[`
+      printValue(item);
+      output += `],`;
+    });
+  }
+  function printValue(value: any) {
+    if (Array.isArray(value)) {
+      printArray(value);
+    } else if (typeof value === 'object') {
+      printObject(value);
+    } else {
+      output += `${typeof value === 'string' ? `'${value}'` : value},` ;
+    }
+  }
+  output += '['
+  printValue(args);
+  output += ']'
+
+  // convert [10,[["dole",1483,],]] -> 10,[["dole",1483]]
+  return output.replace(/,(?=\])/g, '').replace(/^\[|\]$/g, '');
```

Comment:
Can you explain this block? Based on the comment, it looks like `args` is surrounded by a `[]`, which is then stripped by the regex.

Reply:
I reworked the function and removed the regex.